pootle/apps/pootle_language/models.py

```diff
@@ -76,9 +78,20 @@ def pootle_path(self):
         return '/%s/' % self.code
 
     @property
+    def iso_name(self):
+        """ISO full name for the language."""
+        lang_code = simplify_to_common(self.code)
```

Comment:
Just added more code in https://github.com/translate/translate/pull/3636 to handle `en_ZA` and `es-ES` by retrieving separatedly the language name and the country name. But for `ca@valencia` I am afraid it is not possible to do anything, so it falls back to language fullname as defined in Pootle database.